rfcs/2025-carousel.md

```diff
@@ -0,0 +1,480 @@
+
+Notably, this pattern is absent from most major component libraries, largely due to its controversial nature and the unique accessibility challenges it poses. Dedicated third-party implementations have opted to emulate scrolling through transforms and JavaScript-driven snapping. While effective, these approaches can significantly inflate bundle size - oftentimes exceeding 10KB minified+gzipped — and, more importantly, will neither feel fully native nor offer adequate accessibility.
+
+We have a unique opportunity to leverage React Aria’s hook-based API and layout platform to deliver an accessible rotation controller that aligns with RAC's composition architecture and is extremely versatile. Unlike existing solutions, such an implementation could provide a truly native experience, with minimal bundle overhead, potentially achieving an order-of-magnitude improvement not only in bundle size but also in feature scope and accessibility.
+
+## Example
+
+```tsx
+import {Button, Carousel, Slide, SlidePicker, SlidePickerList, SlideShow} from 'react-aria-components';
+import {ChevronLeft, ChevronRight, Dot, Play} from 'lucide-react';
+
+<Carousel>
+  <Button slot="play"><Play size={16}/></Button>
+  <Button slot="previous"><ChevronLeft size={16}/></Button>
+  <Button slot="next"><ChevronRight size={16}/></Button>
+
+  <SlidePickerList>
+    <SlidePicker id="1"><Dot size={8}/></SlidePicker>
+    <SlidePicker id="2"><Dot size={8}/></SlidePicker>
+    <SlidePicker id="3"><Dot size={8}/></SlidePicker>
+  </SlidePickerList>
+
+  <SlideShow>
+    <Slide id="1">Slide 1</Slide>
+    <Slide id="2">Slide 2</Slide>
+    <Slide id="3">Slide 3</Slide>
+  </SlideShow>
+</Carousel>
+```
+<details >
+  <summary style="font-weight: bold">Show CSS</summary>
+  <div style="background-color: #313640; white-space: pre;">
+    .react-aria-Carousel {
+      display: grid;
+      grid-template-areas:
+        "play . previous next"
+        "show show show show"
+        "picker picker picker picker";
+      grid-template-rows: repeat(3, max-content);
+      grid-template-columns: auto 1fr auto auto;
+
+      .react-aria-Button[slot="play"] {
+        grid-area: play;
+      }
+
+      .react-aria-Button[slot="previous"] {
+        grid-area: previous;
+      }
+
+      .react-aria-Button[slot="next"] {
+        grid-area: next;
+      }
+
+      .react-aria-SlidePickerList {
+        grid-area: picker;
+        display: flex;
+        gap: 4px;
+      }
+
+      .react-aria-SlideShow {
+        grid-area: show;
+        display: flex;
+        gap: 12px;
+      }
+    }
+  </div>
+</details>
+
+## Features
+
+A carousel can be built using `<div role="tab">` and `<div role="tabpanel">` HTML elements, but this only supports one perceivable element at a time. Carousel helps you build accessible multi-view rotation components that can be styled as needed.
```

Comment:
something that isn't quite clear to me is whether or not a "presentational" carousel is truly a carousel or just a horizontally scrolling cardview/listbox/gridlist. I suppose this is where the screenreader announcements and support for the various carousel controls may play a role in differentiating them, but I'll need to check the aria pattern.